llvm/lib/Target/PowerPC/PPCISelLowering.cpp

```diff
@@ -7291,7 +7294,21 @@ SDValue PPCTargetLowering::LowerFormalArguments_AIX(
       SDValue FIN = DAG.getFrameIndex(FI, PtrVT);
       SDValue ArgValue =
           DAG.getLoad(ValVT, dl, Chain, FIN, MachinePointerInfo());
-      InVals.push_back(ArgValue);
+
+      // While the ABI specifies that the higher bits of the load should be
```

Comment:
real minor nit - a suggested slight rewording -
```
// While the ABI specifies the argument type is sign or zero extended out to register width 
// not all code is compliant. We truncate and re-extend to be more forgiving of these callers
// when the argument type is smaller then register width.
```

Reply:
More typo/grammar fixes:
While the ABI specifies the argument type is (sign or zero) extended out to register width, not all code is compliant. We truncate and re-extend to be more forgiving of these callers when the argument type is smaller than register width.